### PR TITLE
Remove 33 unused Boost includes

### PR DESCRIPTION
**Pull request by practicalswift**
Remove 33 unused Boost includes.

### CODE REVIEW

File: src/util.cpp [hunk 1]
```diff
@@ -77,11 +77,7 @@
 #endif
 
 #include <boost/algorithm/string/case_conv.hpp> // for to_lower()
-#include <boost/algorithm/string/join.hpp>
-#include <boost/algorithm/string/predicate.hpp> // for startswith() and endswith()
```
Comment (fanquake) [inline src/util.cpp]: starts_with() is [still used](https://github.com/practicalswift/bitcoin/blob/d399e1b1cf60577b2f74e1f72393ea132b4c80f4/src/util.cpp#L401).

Comment (practicalswift) [inline src/util.cpp]: Good catch! Fixing!

**Review by dcousens — APPROVED** [review]

**Comment by paveljanik** [comment]
compiles ACK https://github.com/bitcoin/bitcoin/pull/10546/commits/d652c7515bbae6d51470cf5ee538b0d5cad9547f

**Comment by jonasschnelli** [comment]
Gitian builds this fine: https://bitcoin.jonasschnelli.ch/build/166
ACK https://bitcoin.jonasschnelli.ch/build/166

**Comment by practicalswift** [comment]
Merge conflicts resolved!

**Comment by fanquake** [comment]
Builds are failing with
```
  CXX      wallet/libbitcoin_wallet_a-feebumper.o
In file included from wallet/crypter.cpp:5:0:
wallet/crypter.h:195:5: error: ‘signals2’ in namespace ‘boost’ does not name a type
     boost::signals2::signal<void (CCryptoKeyStore* wallet)> NotifyStatusChanged;
     ^
wallet/crypter.cpp: In member function ‘bool CCryptoKeyStore::Lock()’:
wallet/crypter.cpp:167:29: error: ‘NotifyStatusChanged’ was not declared in this scope
     NotifyStatusChanged(this);
                             ^
wallet/crypter.cpp: In member function ‘bool CCryptoKeyStore::Unlock(const CKeyingMaterial&)’:
wallet/crypter.cpp:205:29: error: ‘NotifyStatusChanged’ was not declared in this scope
     NotifyStatusChanged(this);
```

**Comment by practicalswift** [comment]
@fanquake Thanks for the notification. Builds now passing!

**Comment by fanquake** [comment]
utACK 49de096

**Comment by sipa** [comment]
utACK 49de096c2adcfaf15f99a3355dc4e22f79d56928

**Pull request merged**